### DISCUSSION TITLE
Tarcer RankLimit

### DISCUSSION THREAD

**Discussion by huansouza** [FeynCalc/feyncalc#77] [Q&A] [answered]
Hi,

I'm trying to compute some diagrams at two-loops using FeynCalc+Tarcer. However, the output is  the following:
![Screenshot_20201029_191146](https://user-images.githubusercontent.com/62699121/97638234-e4800880-1a1a-11eb-8bc2-cc63b6df427c.png)
I believe that the problem here is the RankLimit, which I'm using {2, 6}. I don't know what should the RankLimit be in this case. Could anyone help me, please?

Cheers,

Huan.

**Comment by huansouza** [comment]
By the way, here is the code that I'm using. I'm using basically the same code that is shown in the two-loops example on
 the FeynCalc page.

[Tarcer_integrals.tar.gz](https://github.com/FeynCalc/feyncalc/files/5465220/Tarcer_integrals.tar.gz)


Cheers,

Huan.

**Comment by vsht** [comment]
With the "standard" `RankLimit` equal to `{2,5}` I get a simpler list of masters

```
{FAD[{k1, m1}, {k1 - p, m1}], 
 FAD[k1, {k2, m1}, k1 - p, {k2 - p, m1}, {k1 - k2, m1}], 
 FAD[{k1, m1}, k2, {k1 - p, m1}, k2 - p, {k1 - k2, m1}], 
 FAD[{k1, m1}, {k1 - k2, m1}, k2 - p], 
 FAD[{k1, m1}, {k1, m1}, {k1 - k2, m1}, k2 - p]}
```

[Tarcer_integrals.zip](https://github.com/FeynCalc/feyncalc/files/5471516/Tarcer_integrals.zip)

However, personally I never use Tracer for anything, so I'm not familiar with the specifics of tuning the `RankLimit`.
What I mostly employ is FIRE, but not via the FeynHelpers interface (which is more like a toy) but using some 
private codes.

Cheers,
Vladyslav

**Reply by huansouza** [reply]
Thank you for your help, this is already valuable.

I was looking more carefully the file TARCERSource.m and I believe that the problem is not the RankLimit, but the base integrals that are already implemented. Then, I think that I'll need to compute the remaining integrals by myself and include them in my TARCERSource.m file. Despite that, I'll take a look at FIRE, thanks.

Best wishes,

Huan.

**Comment by vsht** [comment, marked as answer]
Do you mean the base integrals from $BasisIntegrals?
Normally you always need to compute the masters separately, since this is not the task of an IBP reduction tool.

**Reply by vsht** [reply]
The massless 2-point functions are of course known very well even at 4-loops (cf. e.g. in the FORCER code).
On-shell 2L 2-point integrals with one mass are also well known. If you have more masses involved, one has
to check with the literature what is already analytically known and what not.

**Reply by huansouza** [reply]
Yes, I was talking about the base integrals from $BasisIntegrals.

I will follow your suggestion and look for the results of the base integrals in the literature. In fact, yesterday I  was talking with a collaborator about this problem and we believe that the best solution is to compute the masters separately.

Thank you so much for your help.

**Reply by vsht** [reply]
Ok, if it's only about masters, there is probably not much point to use FIRE. It doesn't contain any explicitly evaluated integrators.

However, I don't think that you need to compute anything separately, unless it's meant for learning purposes.
From the way your diagrams look like (off-shell, one mass scale) you can probably find everything in

https://arxiv.org/pdf/hep-ph/9808242.pdf

If it's for pheno purposes anyway and pure analytic results are not necessary, just use TSIL

https://www.niu.edu/spmartin/TSIL/

**Reply by huansouza** [reply]
Oh, it seems to me that all that I need is in your first reference. Thank you so much for that, this is gonna be very helpful.

I'm having some trouble understanding the outputs of the final results, but I'll start another topic.
